relayer/src/transfer.rs

```diff
@@ -71,6 +75,57 @@ impl FromStr for Amount {
     }
 }
 
+#[derive(Copy, Clone)]
+pub enum TransferTimeout {
+    Height(Height),
+    Timestamp(Timestamp),
+    HeightAndTimestamp(Height, Timestamp),
+}
+
+impl TransferTimeout {
+    pub fn new(
+        timeout_height_offset: u64,
+        timeout: Duration,
+        destination_chain_status: &ChainStatus,
+    ) -> Result<Self, TransferError> {
+        let offset_is_zero = timeout_height_offset == 0;
+        let timeout_is_zero = timeout == Duration::ZERO;
+
+        let timeout_height = destination_chain_status.height.add(timeout_height_offset);
+
+        let timeout_timestamp = (destination_chain_status.timestamp + timeout)
```

Comment:
Here I updated the timeout timestamp to be computed from the destination chain's timestamp rather than the relayer's local time.

I believe I haven't mistaken the ordering, and it should be the destination chain, not the source chain's timestamp, correct?

Reply:
cc @hu55a1n1 can you take a look at this, please?